### PR TITLE
C#: Add assembly attributes to assemblies built with Bazel

### PR DESCRIPTION
**Pull request by tamasvajk**
This PR cleans up the build configuration for C# projects:

- Removes the last `AssemblyInfo.cs` files and moves `InternalsVisibleToAttributes` to csproj files.
- Removes the corresponding `"Properties/*.cs"` source declarations from Bazel build scripts.
- Ensures that assembly description attributes are added to the output artefacts when they are built with Bazel.

### CODE REVIEW

**Review by criemen — APPROVED** [review]
LGTM, but I think someone from the C# team should also review this PR.

File: csharp/scripts/gen-assembly-info.py [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
 
-[assembly: AssemblyInformationalVersion("{version_string}")]
+[assembly: AssemblyTitle("{opts.name}")]
+[assembly: AssemblyProduct("CodeQL")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
+[assembly: AssemblyCompany("GitHub")]
+[assembly: AssemblyCopyright("Copyright © 2024 GitHub")]
+
+[assembly: System.Runtime.Versioning.TargetFramework(".NETCoreApp,Version=v8.0", FrameworkDisplayName = ".NET 8.0")]
```
Comment (michaelnebel) [inline csharp/scripts/gen-assembly-info.py:31]: Should we record in the C# wiki that this requires updating when we upgrade .NET?

**Review by michaelnebel — APPROVED** [review]
Looks plausible to me!

File: csharp/scripts/gen-assembly-info.py (hunk 1 above)
Comment (tamasvajk) [reply csharp/scripts/gen-assembly-info.py:31]: Good point. I'm doing it.

**Pull request merged**